### DISCUSSION TITLE
Parsing error: cannot find module 'next/babel'

### DISCUSSION THREAD

**Discussion by thomasweigert91** [neuefische/questions#333] [Web] [answered]
Hey,

I get a red line under import all the time when importing from react with the error described above. I found out that I need to add "next/babel" in the eslintrc at "extends". This resolves the error message, but I can't import anything. Whether I type import {React} or import {styled}, it finds nothing and doesn't even complete the import for me, which has always worked under React otherwise. With Next it somehow doesn't work. Also reinstalling the repo followed by npm install does nothing.
<img width="750" alt="Bildschirmfoto 2022-11-28 um 14 39 45" src="https://user-images.githubusercontent.com/114309234/204292736-75097e2c-163c-4559-95e5-b2800e6ecf43.png">

**Comment by actyralf** [comment]
You don't need `import React from "react"` anymore, so the linter complains.
Please try to remove the line and to remove the additional babel config you did in `.eslintrc`.

**Comment by pixelass** [comment, marked as answer]
Next.js works with SWC now. It is a faster replacement for babel, therefore babel should not be required at all. (please remove any `.babelrc`, `babel.config.js`, `.babelrc.js`, `.babelrc.json` or `babel` key in `package.json`)

## What is babel?

[Babel](https://babeljs.io/) transforms JSX to JS. (`<div/>` -> `React.createElement("div")`)

## What is SWC?

[SWC](https://swc.rs/) does what babel does but it is faster since it is written in [Rust](https://www.rust-lang.org/) (faster operations).

Conclusion: Babel should not be needed.


I created a boilerplate for nextjs with examples for all integrations: 

* Eslint: https://github.com/neuefische/capstone-next-template/blob/main/.eslintrc.json#L13-L14
* SWC: https://github.com/neuefische/capstone-next-template/blob/main/.swcrc#L1-L23

I hope this helps solve your problem. Feel free to contact me on slack.

Further reading: 

* https://swc.rs/docs/migrating-from-babel
* https://nextjs.org/docs/advanced-features/compiler#why-swc

**Comment by thomasweigert91** [comment]
Thank you for the answers. It did not work. I reinstalled everything the third time and, don't ask me why, it works now...
